sdks/go/pkg/beam/runners/prism/internal/README.md

```diff
@@ -37,31 +39,505 @@ Handler configurations are registered by dependant packages.
 
 `urns` contains beam URN strings pulled from the protos. Leaf package.
 
-`engine` contains the core manager for handling elements, watermarks, and windowing strategies.
-Determines bundle readiness, and stages to execute. Leaf package.
+`engine` contains the core manager for handling elements, watermarks, windowing strategies,
+state, and timers. Determines bundle readiness, and stages to execute. Leaf package.
+Should never depend on the protocol buffers.
 
 `jobservices` contains GRPC service handlers for job management and submission.
 Should only depend on the `config` and `urns` packages.
 
 `worker` contains interactions with FnAPI services to communicate with worker SDKs. Leaf package
 except for dependency on `engine.TentativeData` which will likely be removed at some point.
 
-`internal` AKA the package in this directory root. Contains fhe job execution
-flow. Jobs are sent to it from `jobservices`, and those jobs are then executed by coordinating
-with the `engine` and `worker` packages, and handlers urn.
+`internal` AKA the package in this directory root. Contains the job execution
+flow. Jobs are sent to it from `jobservices`, through the Job Executor function,
+and those jobs are then executed by coordinating with the `engine` and `worker` packages.
+Handles the pipeline protos directly.
 Most configurable behavior is determined here.
 
 `web` contains a web server to visualize aspects of the runner, and should never be
 depended on by the other packages for the runner.
 It will primarily abstract access to pipeline data through a JobManagement API client.
 If no other option exists, it may use direct access to exported data from the other packages.
 
-# Testing
+## Testing
 
 The sub packages should have reasonable Unit Test coverage in their own directories, but
-most features will be exercised via executing pipelines in this package.
+most features will be exercised via executing pipelines in the internal package, and engine
+package. To avoid overly re-implementing tests, use of the "_test" formalism to validate
```

Comment:
```suggestion
most features will be exercised via executing pipelines in the internal and engine
packages. To avoid overly re-implementing tests, use of the "_test" formalism to validate
```